django/db/models/expressions.py

```diff
@@ -689,6 +689,16 @@ def as_sql(self, compiler, connection):
     def get_group_by_cols(self, alias=None):
         return [self]
 
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None,
+                           summarize=False, for_save=False, simple_col=False):
+        other_fields = (set(query.model._meta.get_fields()) -
+                        set(query.model._meta.get_fields(include_parents=False)))
+        for field in other_fields:
+            if field.name in self.sql:
```

Comment:
This solution introduce really unexpected behavior, i.e. change every raw SQL that contains name of any column from a parent model with that column, e.g. if a parent model contains column `name` then following examples will be replaced by `"annotations_store"."name"`:
 - `case when name='foo' then 'oof' else 'foo' end` -> `"annotations_store"."name"`,
 - `concat(chain, 'name')` -> `"annotations_store"."name"`,
 - `other_column_with_name_in_it` -> `"annotations_store"."name"`, etc.

Reply:
I just wanted to add join query through `resolve_ref`. `return`'s place was wrong. the challenge is separating the columns from the string. If we can take the columns properly, we can handle other ops. How should we proceed?

Reply:
`regex` will be clunky. IMO unnecessary `JOIN`'s are acceptable in this case, there is not much we can do.